packages/lib/src/toolbar/controls/Selector/Option.tsx

```diff
@@ -0,0 +1,35 @@
+import type { UseInteractionsReturn } from '@floating-ui/react';
+import { useListItem } from '@floating-ui/react';
+import type { PropsWithChildren } from 'react';
+
+import selectorStyles from './Selector.module.css';
+
+interface Props {
+  activeIndex: number | null;
+  selectedIndex: number;
+  getItemProps: UseInteractionsReturn['getItemProps'];
+  onSelect: (index: number) => void;
+}
+
+function Option(props: PropsWithChildren<Props>) {
+  const { activeIndex, selectedIndex, children, getItemProps, onSelect } =
+    props;
+  const { ref, index } = useListItem();
```

Comment:
This hook works together with `FloatingList`. It retrieves the index of the option based on rendering order (regardless of whether they are part of an option group) and registers it in the `listRef` array (cf. `Selector`) so `useListNavigation` can navigate between options.